services/datamanager/datasync/sync.go

```diff
@@ -124,45 +120,47 @@ func NewManager(logger golog.Logger, partID string, client v1.DataSyncServiceCli
 func (s *syncer) Close() {
 	s.closed.Store(true)
 	s.cancelFunc()
-	if s.conn != nil {
-		if err := s.conn.Close(); err != nil {
-			s.logger.Errorw("error closing datasync server connection", "error", err)
-		}
-	}
 	s.backgroundWorkers.Wait()
 	close(s.syncErrs)
 	s.logRoutine.Wait()
 	//nolint:errcheck
 	_ = s.logger.Sync()
+	if s.conn != nil {
```

Comment:
Why this moved?

Reply:
This needs to occur after all the upload routines have returned/been cancelled (`s.backgroundWorkers.Wait()` returns), because otherwise those in progress routines might still be trying to use this connection. This results in those erroring with `send on closed connection` (maybe not exact error because I can't find it in my bash history - but something like that)

Reply:
Ahhh got it - thank you!